cdap-api/src/main/java/co/cask/cdap/api/Admin.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api;
+
+import co.cask.cdap.api.dataset.DatasetManagementException;
+import co.cask.cdap.api.dataset.DatasetProperties;
+
+import javax.annotation.Nullable;
+
+/**
+ * This interface provides methods for operational calls from within a CDAP application.
+ */
+public interface Admin {
```

Comment:
I think having all admin methods in a single Admin interface would make Admin interface huge. Instead we can have Admin return DatasetAdmin, and DatasetAdmin would define all these dataset admin methods. We can then add ProgramAdmin and other admins to Admin interface.

What do you say?